synfig-core/src/modules/mod_svg/svg_parser.cpp

```diff
@@ -2113,3 +2063,185 @@ get_inkscape_version(const xmlpp::Element* svgNodeElement)
 	}
 	return 0; // not inkscape ;)
 }
+
+void
+Style::merge(const xmlpp::Element *elem)
+{
+	if (elem->get_name() == "style")
+		return;
+
+	// stores the style defined for this element
+	// lower priority: style property values defined by presentation-kind attributes
+	// medium priority: those values from CSS selectors (TODO)
+	// highest priority: those values in "style" attribute contents
+
+	Style style;
+
+	style.merge_presentation_attributes(elem);
+
+	// now CSS info - TODO
+
+	// and then "style" attribute
+	std::string style_str = elem->get_attribute_value("style");
+
+	if (!style_str.empty()) {
+		style.merge_style_string(style_str);
+	}
+
+	// merge properties with current style
+	// Normally, we just replace them with new values
+	// However, there are some special cases (SVG 1.1 6.14):
+	//   - relative values will have the same units as the value to which it is relative
+	//      and percentage lengths are relative to current viewport or bounding box dimensions
+	//   - if a 'clip-path' property is specified on an ancestor element, and the current element
+	//       has a 'clip-path' of none, the ancestor's clipping path still applies to current element
+
+	for (auto prop : style.data) {
+		if (prop.first == "clip-path" && prop.second == "none")
+			continue;
+		// TODO: Relative values?
+		data[prop.first] = prop.second;
+	}
+}
+
+std::string
+Style::get(const std::string &property, std::string default_value) const
+{
+	auto item = data.find(property);
+	if (item == data.end() || item->second.empty())
+		return default_value;
+	return item->second;
+}
+
+double
+Style::compute(const std::string &property, std::string default_value, double reference_value) const
+{
+	std::string value = get(property, default_value);
+
+	if (value.empty()) {
+		// It would only happen if default_value is empty, as push() forbids empty value
+
+		error("Layer_Svg: %s",
+			  etl::strprintf(_("Value to be computed for '%s' is empty. Assuming zero. Internal error."), property.c_str()).c_str());
+		return 0;
+	}
+
+	try {
+		if (value.back() != '%')
+			return std::stod(value);
+		else
+			return reference_value * std::stod(value) / 100.;
+	} catch(...) {
+	}
+
+	if (value != default_value) {
```

Comment:
There might still be a problem with the empty default value here. For example, if you push a wrong value into a property and pass an empty default. 

Also if you add a test for empty value here, when code begins to repeat yourself.
Here my proposed change:

https://github.com/ice0/synfig/commit/2e105a38ccd09bcf0014621e12e6bc46543cf545

Reply:
done! Thank you :D